staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_aggregator.go

```diff
@@ -132,36 +166,46 @@ func sortByPriority(specs []openAPISpecInfo) {
 		groupPriorities: map[string]int32{},
 	}
 	for _, spec := range specs {
+		if spec.apiService.Spec.Service == nil {
+			continue
+		}
 		if pr, found := b.groupPriorities[spec.apiService.Spec.Group]; !found || spec.apiService.Spec.GroupPriorityMinimum > pr {
 			b.groupPriorities[spec.apiService.Spec.Group] = spec.apiService.Spec.GroupPriorityMinimum
 		}
 	}
 	sort.Sort(b)
 }
 
-// buildOpenAPISpec aggregates all OpenAPI specs.  It is not thread-safe.
+// buildOpenAPISpec aggregates all OpenAPI specs.  It is not thread-safe. The caller is responsible to hold proper locks.
```

Comment:
then this should probably be `buildOpenAPISpec_locked`

Reply:
is using "_" in names recommended? `buildOpenAPISpec` is a private method anyway. it only used in this type.

Reply:
golint hates underscores